apps/state/lib/state/shape.ex

```diff
@@ -24,29 +24,53 @@ defmodule State.Shape do
   Select shapes provided a list of route ids and direction id.
   """
   @spec select_routes([Route.id()], Direction.id() | nil) :: [Shape.t()]
-  def select_routes(route_ids, direction_id) do
+  def select_routes(route_ids, direction_id, api_version \\ "2019-07-01") do
     opts =
       case direction_id do
         id when id in [0, 1] -> %{routes: route_ids, direction_id: id}
         _ -> %{routes: route_ids}
       end
 
-    opts
-    |> State.Trip.filter_by()
-    |> Enum.map(& &1.shape_id)
-    |> Enum.uniq()
-    |> by_ids()
-    |> Enum.sort_by(&{-&1.priority, &1.name})
+    shapes =
+      opts
+      |> State.Trip.filter_by()
+      |> Enum.map(& &1.shape_id)
+      |> Enum.uniq()
+      |> by_ids()
+      |> Enum.sort_by(&{-&1.priority, &1.name})
+
+    if api_version < "2019-07-01" do
```

Comment:
The `state` module doesn't know anything about the API version, and it's better to keep it that way.

Instead, you can override `def name(shape, conn)` in ShapeView, and get the API version there.